spectator/writer/writer.go

```diff
@@ -0,0 +1,25 @@
+package writer
+
+import "fmt"
+
+// Writer that accepts SpectatorD line protocol.
+type Writer interface {
+	Write(line string) // TODO check type
+}
+
+// TODO consider extracting common logging logic into a default trait that is then included in all implementations
+
+type PrintWriter struct {
+}
+
+func (p *PrintWriter) Write(line string) {
+	fmt.Print(line)
```

Comment:
This is more of a logging implementation,, but I like it as a starting point.

The PrintWriter in the Python implementation writes to a TextIO instance (file).

https://github.com/Netflix/spectator-py/blob/main/spectator/sidecarwriter.py#L106

This is what allows it to either log output to disk, or use the Unix Domain Socket, which is a file-based implementation of IPC.

Reply:
Yup, that was the first I wrote just to make it compile. Then I realized I needed the MemoryWriter for tests.

I'll revisit as I implement the real writers. Might delete it or turn it into a Logger implementation